accord-core/src/main/java/accord/coordinate/CheckOn.java

```diff
@@ -136,21 +138,28 @@ class OnDone implements MapReduceConsume<SafeCommandStore, Void>
 
         public OnDone()
         {
-            Ranges localRanges = node.topology().localRangesForEpochs(txnId.epoch, untilLocalEpoch);
-            PartialRoute<?> selfRoute = route().slice(localRanges);
+            Ranges sliceRanges = node.topology().localRangesForEpochs(txnId.epoch(), untilLocalEpoch);
+            PartialRoute<?> selfRoute = route().slice(sliceRanges, Maximal);
+            if (selfRoute.domain().isRange())
+            {
+                // for range queries, we want to "cover" the entirety of every range
```

Comment:
We want `covers` to cover the bit we're slicing and the `Route` so that our invariants hold. We're only slicing here though, so not really doing any extra work.

Perhaps later we can look into slicing the `Route` to a `Minimal` one so we only seek information from the minimum number of replicas to answer our own local queries. I will add a big TODO.

Reply:
We talked about this and my take away is that this looks like it should be a no-op because self `sliceRanges` should be a superset of `selfRoute`'s ranges, because `selfRoute` is a `route()` sliced on `sliceRanges`, but I see in the original code `slice` on `route()` was called with `Maximal` and then `Maximal` was removed and it now calls `RangeRoute.slice` which calls `AbstractRanges.slice` with `Overlapping`.

So `selfRoute` slices with `Overlapping` includes every range in the route that overlaps with any of `sliceRanges`.

Where this seems odd is that the response being given doesn't actually cover the entirety of the range from the route?  The whole purpose of tracking coverage is to preserve that knowledge so we can see what is and isn't covered and this bypasses that.

Reply:
Ok, so the issue here is that we do not truncate the ranges in the `Route` we pass to `reconstitutePartial`, so when we slice the `partialTxn` to `sliceRanges` before invoking  `reconstitutePartial`, `reconstitutePartial` says we don't cover the whole range.

But, I think this is a problem with how this is modelled. I will give it some thought, perhaps for range transactions we should not be providing a `Route` to reconstitute partial.

Reply:
Ok, I've addressed this by separately slicing the keys/key ranges and the ranges they cover.